core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -58,6 +60,7 @@
   private Long existingRowsCount = null;
   private Long deletedRowsCount = null;
   private PartitionFieldSummary[] partitions = null;
+  private byte[] keyMetadata = null;
```

Comment:
A nit: We don't have to initialize a variable in Java, but since all others do the same thing. It should be fine to be consistent here.

Reply:
I think it's better to initialize than to let it be implicit. That way it is always clear to anyone reading the code what the initial value is, whether or not they know what the JVM would fill in.

Reply:
Either way is OK to me as long as it is consistent here or across the project. To avoid getting back to this style issue again(we can work on more important things with that energy), I suggest to have this kind of style preference written down as a developer/reviewer guidance, so that at least we can refer to it.